npm/translatePolicy.go

```diff
@@ -0,0 +1,1035 @@
+// Copyright 2018 Microsoft. All rights reserved.
+// MIT License
+package npm
+
+import (
+	"github.com/Azure/azure-container-networking/log"
+	"github.com/Azure/azure-container-networking/npm/iptm"
+	"github.com/Azure/azure-container-networking/npm/util"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type portsInfo struct {
+	protocol string
+	port     string
+}
+
+func craftPartialIptEntrySpecFromPort(portRule networkingv1.NetworkPolicyPort, sPortOrDPortFlag string) []string {
+	partialSpec := []string{}
+	if portRule.Protocol != nil {
+		partialSpec = append(
+			partialSpec,
+			util.IptablesProtFlag,
+			string(*portRule.Protocol),
+		)
+	}
+
+	if portRule.Port != nil {
+		partialSpec = append(
+			partialSpec,
+			sPortOrDPortFlag,
+			portRule.Port.String(),
+		)
+	}
+
+	return partialSpec
+}
+
+func craftPartialIptablesCommentFromPort(portRule networkingv1.NetworkPolicyPort, sPortOrDPortFlag string) string {
```

Comment:
sPortOrDPortFlag isn't used here

Reply:
Let's keep it this way for now. There could be upstream specification changes.